src/main/java/org/opentripplanner/middleware/otp/OtpRequestProcessor.java

```diff
@@ -13,81 +13,121 @@
+        // available at (depending on config) http://localhost:4567/otp/*
+        spark.get(OTP_ROOT_ENDPOINT + "/*", OtpRequestProcessor::proxy);
+    }
+
+    /**
+     * Inspect all requests that are made to OTP. If the request is of interest (for instance plan response) intercept
+     * the response from OTP and process accordingly. In all cases, pass the response back to the requester.
+     */
+    private static String proxy(Request request, spark.Response response) {
         if (OTP_SERVER == null) {
             logMessageAndHalt(request, HttpStatus.INTERNAL_SERVER_ERROR_500, "No OTP Server provided, check config.");
             return null;
         }
 
-        String batchId = request.queryParams("batchId");
-        if (batchId == null) {
-            //TODO place holder for now
-            batchId = "-1";
-        }
-
         // attempt to get response from OTP server based on requester's query parameters
-        OtpDispatcher otpDispatcher = new OtpDispatcherImpl(OTP_SERVER);
-        OtpDispatcherResponse otpDispatcherResponse = otpDispatcher.getPlan(request.queryString(), OTP_SERVER_PLAN_END_POINT);
+        OtpDispatcherResponse otpDispatcherResponse = OtpDispatcher.serviceRequest(request.queryString(), request.uri());
         if (otpDispatcherResponse == null || otpDispatcherResponse.responseBody == null) {
             logMessageAndHalt(request, HttpStatus.INTERNAL_SERVER_ERROR_500, "No response from OTP server.");
             return null;
         }
 
+        String targetService = extractTargetService(request.uri());
+        // if the target service is plan, process response
+        if (targetService.equalsIgnoreCase(OTP_PLAN_ENDPOINT)) plan(request, otpDispatcherResponse);
+
+        // provide response to requester as received from OTP server
+        response.type("application/json");
+        response.status(otpDispatcherResponse.statusCode);
+        return otpDispatcherResponse.responseBody;
+    }
+
+    /**
+     * Extract from the requesting URI the OTP target service e.g. 'routers/default/plan'. The assumption is that the
+     * target service will be the remaining part of the URL after the OTP root endpoint has been removed.
+     *
+     * The following example is based on calls being made to the FDOT OTP server:
+     *
+     * Middleware call: http://otp-middleware.com/otp/routers/default/plan
+     * FDOT OTP call: https://fdot-otp-server.com/otp/routers/default/plan
+     * OTP root endpoint: /otp/
+     * Target service: routers/default/plan
+     */
+    private static String extractTargetService(String uri) {
+        return uri.replace(OTP_ROOT_ENDPOINT, "");
+    }
+
+    /**
+     * Process plan response from OTP. Store the response if consent is given. Handle the process and all exceptions
+     * seamlessly so as not to affect the response provided to the requester.
+     */
+    private static void plan(Request request, OtpDispatcherResponse otpDispatcherResponse) {
```

Comment:
This method name should be changed. We're not so much planning as we are handling the plan response. So maybe `handlePlanTripResponse`?

Reply:
Updated.